src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -819,6 +820,8 @@ private void joinTokenRing(int delay) throws ConfigurationException
         boolean dataAvailable = true; // make this to false when bootstrap streaming failed
         if (shouldBootstrap())
         {
+            setMode(Mode.WAITING_TO_BOOTSTRAP, "Awaiting start bootstrap call", true);
```

Comment:
Does this block new clusters from starting up?

Reply:
Nop, because we only wait for the signal after the `shouldBootstrap()` call, which is false for new clusters because the nodes are in the seeds